code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -76,20 +76,20 @@
 	venom_per_bite = 10
 	move_to_delay = 5
 
-/mob/living/simple_animal/hostile/poison/giant_spider/hunter/handle_automated_action()
+/mob/living/simple_animal/hostile/poison/giant_spider/handle_automated_action()
 	if(!..()) //AIStatus is off
-		return
+		return 0
 	if(AIStatus == AI_IDLE)

```

Comment:
Are you sure this is gonna run? 
handle_automated_action() is (if memory serves) only called when AIStatus == AI_ON
So if handle_automated_action() is being called, can't you assume that AIStatus != AI_IDLE?

Reply:
It should work now; hacky as it is.